lib/puppet/application/agent.rb

```diff
@@ -387,9 +388,19 @@ def onetime(daemon)
 
   def main(daemon)
     Puppet.notice _("Starting Puppet client version %{version}") % { version: Puppet.version }
+    Puppet.debug {"Current config:\n#{get_config}"}
     daemon.start
   end
 
+  def get_config()
+      config_face = Puppet::Face["config", "0.0.1"]
+      result = config_face.send("print",{:section=>"main"})
```

Comment:
nit, prefer to call to the method explicitly instead of `send`:

```ruby
face.print(section: "main")
```

Reply:
Actually since the output produces so many lines and our logging destinations don't handle multiline output very well, it's probably better to log each setting, and skip the sensitive ones:

```ruby
face.print(section: "main").each_pair do |k,v|
  next if k == "http_proxy_password" # etc
  Puppet.debug{ "Setting #{k}=#{v}" }
end

Reply:
I will fix this. Should we return from function if debug not enabled to skip config reading/parsing